index.js

```diff
@@ -428,6 +444,34 @@ router.post("/login", (req, res) => {
 	res.redirect(pathname+"gallery");
 });
 
+router.get("/oidc/redirect", (req, res) => {
+  if (!oidcClient) return error(req, res, "OIDC not configured");
+  const state = Math.random().toString(36).slice(2, 7);
+  oidcStateMap.set(req.headers["X-Forwarded-For"] || req.ip, state);
```

Comment:
Instead of blindly reading X-Forwarded-For, use `app.set("trust proxy")` on webserver init and then just use `req.ip` directly, more info: https://expressjs.com/en/guide/behind-proxies.html